server/aws-lsp-codewhisperer/src/shared/amazonQServiceManager/configurationUtils.ts

```diff
@@ -108,6 +108,14 @@ export async function getAmazonQRelatedWorkspaceConfigs(
     lsp: Lsp,
     logging: Logging
 ): Promise<Readonly<Partial<AmazonQWorkspaceConfig>>> {
+    const clientParams = lsp.getClientInitializeParams()
+    const supportsWorkspaceConfiguration = clientParams?.capabilities?.workspace?.configuration !== false
+
+    if (!supportsWorkspaceConfiguration) {
+        logging.debug('Client does not support workspace configuration, returning default config.')
+        return {}
+    }
```

Comment:
This check was added because the server always sends `workspace/configuration` requests to the client but `ts-lsp-client` doesn't support it which results in failures due to out-of-order messages.

Real clients should use `true` or `undefined` for `workspace.configuration` in client capabilities.